src/sdp/DefaultSDP.ts

```diff
@@ -218,41 +218,75 @@ export default class DefaultSDP implements SDP {
       )
     );
     const srcLines: string[] = this.lines();
-    const dstLines: string[] = [];
+    const fmtpAttribute = DefaultSDP.findOpusFmtpAttribute(srcLines);
+    const dstLines = DefaultSDP.updateOpusFmtpAttribute(srcLines, fmtpAttribute, [
+      `maxaveragebitrate=${maxAverageBitrate}`,
+    ]);
+    return DefaultSDP.linesToSDP(dstLines);
+  }
+
+  withStereoAudio(): DefaultSDP {
+    const srcLines: string[] = this.lines();
+    const fmtpAttribute = DefaultSDP.findOpusFmtpAttribute(srcLines);
+    const dstLines = DefaultSDP.updateOpusFmtpAttribute(srcLines, fmtpAttribute, [
+      'stereo=1',
+      'sprop-stereo=1',
+    ]);
+    return DefaultSDP.linesToSDP(dstLines);
+  }
+
+  static findOpusFmtpAttribute(sdpLines: string[]): string | null {
     const opusRtpMapRegex = /^a=rtpmap:\s*(\d+)\s+opus\/48000/;
     let lookingForOpusRtpMap = false;
     let fmtpAttribute: string | null = null;
-    for (const line of srcLines) {
+    for (const line of sdpLines) {
       if (line.startsWith('m=audio')) {
         lookingForOpusRtpMap = true;
-        fmtpAttribute = null;
-      }
-      if (line.startsWith('m=video')) {
-        lookingForOpusRtpMap = false;
-        fmtpAttribute = null;
       }
       if (lookingForOpusRtpMap) {
         const match = opusRtpMapRegex.exec(line);
         if (match !== null) {
           fmtpAttribute = `a=fmtp:${match[1]} `;
-          lookingForOpusRtpMap = false;
+          break;
```

Comment:
Do we change the logic here? It seems previously we loop through every line in sdp but now we quit after finding the opus line. Does that mean there can be only one opus line?

Reply:
Ah good catch. There can actually be multiple opus lines. I updated the code and unit test to account for this. Also added comments.